### PR TITLE
Aggregate oracle price

### PR DESCRIPTION
**Pull request by Mixa84**
#### What kind of PR is this?:

/kind fix

#### What this PR does / why we need it:
Changes priceFeedId filed in `setcollateraltoken` and `setloantoken` tx to be `token/currency` pair instead of txid of an oracle.

### CODE REVIEW

File: src/masternodes/rpc_loan.cpp [hunk 1]
```diff
@@ -446,7 +460,13 @@ UniValue updateloantoken(const JSONRPCRequest& request) {
...
         loanToken->name = trim_ws(metaObj["name"].getValStr());
     if (!metaObj["priceFeedId"].isNull())
-        loanToken->priceFeedTxid = uint256S(metaObj["priceFeedId"].getValStr());
+    {
+        std::string tmp = trim_ws(metaObj["priceFeedId"].getValStr());
+        if (tmp.length() < MINIMUM_PRICEFEED_LENGTH || tmp.find("/") == tmp.npos)
+            throw JSONRPCError(RPC_INVALID_PARAMETER,"price feed not in valid format - token/currency!");
+        loanToken->priceFeed.first = tmp.substr(0, tmp.find("/"));
+        loanToken->priceFeed.second = tmp.substr(tmp.find("/")+1, tmp.length()-1);
+    }
```
Comment (hidiego) [inline src/masternodes/rpc_loan.cpp:469]: This code is repeated several times, maybe is better to use a function.

File: src/masternodes/rpc_loan.cpp [hunk 2]
```diff
@@ -1,5 +1,7 @@
 #include <masternodes/mn_rpc.h>
 
+#define MINIMUM_PRICEFEED_LENGTH 8
```
Comment (hidiego) [inline src/masternodes/rpc_loan.cpp:3]: DFI/USD for example would throw error, consider lowering this number.

**Review by dcorral — COMMENTED** [review]
The price feeds are not checked against oracles anyhow so you can set loan/collateral tokens with unsupported pairs.

File: src/masternodes/rpc_loan.cpp [hunk 3]
```diff
@@ -29,14 +29,34 @@ UniValue setLoanTokenToJSON(CLoanSetLoanTokenImplementation const& loanToken, DC
...
-    loanTokenObj.pushKV("priceFeedId", loanToken.priceFeedTxid.GetHex());
+    loanTokenObj.pushKV("priceFeedId", loanToken.priceFeed.first + "/" + loanToken.priceFeed.second);
     loanTokenObj.pushKV("interest", ValueFromAmount(loanToken.interest));
 
     UniValue ret(UniValue::VOBJ);
     ret.pushKV(loanToken.creationTx.GetHex(), loanTokenObj);
     return (ret);
 }
 
+PriceFeedPair DecodePriceFeed(const UniValue& value)
```
Comment (ShengguangXiao) [inline src/masternodes/rpc_loan.cpp:40]: Should add some unit test for this function

Comment (bvbfan) [reply src/masternodes/rpc_loan.cpp:40]: All the tests do it.

**Review by ShengguangXiao — APPROVED** [review]

**Pull request merged**